app/models/concerns/foreman_openscap/policy_extensions.rb

```diff
@@ -42,9 +42,11 @@ def hostgroup_ids
     end
 
     def hostgroup_ids=(ids)
+      hostgroup_assets = []
       ids.reject(&:empty?).map do |id|
-        self.assets << assets.where(:assetable_type => 'Hostgroup', :assetable_id => id).first_or_create!
+        hostgroup_assets << assets.where(:assetable_type => 'Hostgroup', :assetable_id => id).first_or_create!
       end
+      self.assets = hostgroup_assets
```

Comment:
@isimluk - I think this should be amended into 0.3.x, as currently you can't update hostgroups.